src/IdEncoding/IdEncoder.php

```diff
@@ -0,0 +1,11 @@
+<?php
+declare(strict_types=1);
+
+namespace EventSauce\IdEncoding;
+
+use EventSauce\EventSourcing\AggregateRootId;
+
+interface IdEncoder
+{
+    public function encodeId(AggregateRootId|string $id): mixed;
```

Comment:
Why is the return type of the interface `mixed`? when would we need another ID than a string?

Reply:
Some might use an integer value for example.